src/hotspot/cpu/x86/x86.ad

```diff
@@ -7122,11 +7146,44 @@ instruct vcastLtoBS(vec dst, vec src, rRegP scratch) %{
   ins_pipe( pipe_slow );
 %}
 
+instruct vcastLtoFD_avx(vec dst, vec src, vec xtmp1, vec xtmp2, rRegI tmp, rFlagsReg cr) %{
+  predicate((UseAVX <= 2 || !VM_Version::supports_avx512dq()) &&
```

Comment:
`!VM_Version::supports_avx512dq()` is equivalent to `UseAVX <= 2 || !VM_Version::supports_avx512dq()` here, isn't it?

Reply:
```
case Op_VectorCastL2X:
   ...
   } else if (!is_integral_type(bt) && !VM_Version::supports_avx512dq()) {
     return false;
   }
```
Overall, It becomes harder and harder to ensure the relevant AD instructions are exhaustive w.r.t. predicates and supported modes. 

I'd expect to see different variations of `is_floating_point_type(Matcher::vector_element_basic_type(n))` and`VM_Version::supports_avx512dq())` to enumerate all important configurations.